crates/cli/src/commands/apply_pattern.rs

```diff
@@ -285,6 +293,22 @@ pub(crate) async fn run_apply_pattern(
                 }
             }
         };
+        if let Some(lang_option) = &arg.language {
+            if is_pattern_file(&pattern) {
```

Comment:
I don't think we need to special case pattern files - we should also not allow overriding the `language` in those cases if there's a conflict.

Similarly I should be able to `grit apply` a `foo.grit` that doesn't have a `language` specified.

Concretely, the purpose of the `--language` option is to change the *default* pattern language from JS to whatever is passed.

Reply:
This is the only remaining change. I'm the tech lead, you don't need @ilevyor to approve.